src/lib.rs

```diff
@@ -85,7 +85,23 @@ pub trait GenerateVerifiable {
 		who: Self::Member,
 		lookup: impl Fn(usize) -> Result<Self::StaticChunk, ()>,
 	) -> Result<(), ()>;
-	// push_members
+
+	/// Introduce a set of new `Member`s into the intermediate value used to build a new `Members`
+	/// value. This function is especially suitable for low domain sizes, as the provided `chunks`
+	/// must be the whole set of chunks available for a domain, such that a lookup `chunks[i]` would
+	/// yield the corresponding chunk for the `i`th key in the intermediate.
+	fn push_members(
```

Comment:
IMO is not worth adding a new method. As the backend already accepts a lookup Fn accepting a range, perhaps you can live with just the `push_members` and remove the single member push?

Reply:
Mmm I don't think so

Reply:
It is not even on crates.io so I guess we can break the interface to improve it. There is plenty of space for improvement (the associated types are improveable as well)